patches/api/0387-Add-EntityMateEvent.patch

```diff
@@ -0,0 +1,141 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Lulu13022002 <41980282+Lulu13022002@users.noreply.github.com>
+Date: Fri, 24 Jun 2022 11:56:32 +0200
+Subject: [PATCH] Add EntityMateEvent
+
+
+diff --git a/src/main/java/io/papermc/paper/event/entity/EntityMateEvent.java b/src/main/java/io/papermc/paper/event/entity/EntityMateEvent.java
+new file mode 100644
+index 0000000000000000000000000000000000000000..de9a8c84772a343b3ee197a65b587673c80d44b0
+--- /dev/null
++++ b/src/main/java/io/papermc/paper/event/entity/EntityMateEvent.java
+@@ -0,0 +1,129 @@
++package io.papermc.paper.event.entity;
++
++import com.google.common.base.Preconditions;
++import org.bukkit.entity.LivingEntity;
++import org.bukkit.entity.Player;
++import org.bukkit.event.Cancellable;
++import org.bukkit.event.HandlerList;
++import org.bukkit.event.entity.EntityEvent;
++import org.bukkit.inventory.ItemStack;
++import org.jetbrains.annotations.NotNull;
++import org.jetbrains.annotations.Nullable;
++
++/**
++ * Called when a Turtle or a Frog meets a new mate.
```

Comment:
I would generalize a bit, and say something like "Called when a mating occurs that results in a pregnancy". Since this is a different breeding mechanic to other mobs that produce offspring.